preinstall.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+echo "Setting up the development environment..."
+
+install_package_linux() {
+  local PACKAGE=$1
+
+  # apt-get for Debian Ubuntu
+  if command -v apt-get &> /dev/null; then
+    if ! command -v "$PACKAGE" &> /dev/null; then
+      echo "Installing $PACKAGE using apt-get..."
+      sudo apt-get update -y
+      sudo apt-get install -y "$PACKAGE"
+    else
+      echo "$PACKAGE is already installed."
+    fi
+
+  # try dnf for Fedora or RedHat
+  elif command -v dnf &> /dev/null; then
+    if ! command -v "$PACKAGE" &> /dev/null; then
+      echo "Installing $PACKAGE using dnf..."
+      sudo dnf install -y "$PACKAGE"
+    else
+      echo "$PACKAGE is already installed."
+    fi
+
+  else
+    echo "No recognized package manager found."
+    echo "Please install $PACKAGE manually."
+    exit 1
+  fi
+}
+
+install_package_mac() {
+  PACKAGE=$1
+  if ! command -v $PACKAGE &> /dev/null; then
+    echo "Installing $PACKAGE on macOS..."
+    if ! brew install "$PACKAGE"; then
+      echo "Failed to install $PACKAGE"
+      exit 1
+    fi
+  else
+    echo "$PACKAGE is already installed."
+  fi
+}
+
+# Detect the operating system
+OS=$(uname -s)
+
+if [[ "$OS" == "Linux" ]] && ! command -v sudo &> /dev/null; then
```

Comment:
could move this into the next if-clause as you are checking $OS == "Linux" twice here

eg.

if [[ "$OS" == "Linux" ]] ; then 
  if ! command -v sudo &> /dev/null ; then
    ...error
  fi
 
  ....rest of the code

Reply:
true!